src/Microsoft.AspNet.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -172,75 +166,37 @@ protected virtual string GenerateUniqueId()
                                                    tagHelperDescriptor.TypeName)
                        .WriteEndMethodInvocation();
 
-                // Execution contexts are a runtime feature.
-                if (!_designTimeMode)
-                {
-                    _writer.WriteInstanceMethodInvocation(ExecutionContextVariableName,
-                                                          _tagHelperContext.ExecutionContextAddMethodName,
-                                                          tagHelperVariableName);
-                }
-
-                // Render all of the bound attribute values for the tag helper.
-                RenderBoundHTMLAttributes(
-                    chunk.Attributes,
-                    tagHelperDescriptor.TypeName,
-                    tagHelperVariableName,
-                    tagHelperDescriptor.Attributes,
-                    htmlAttributeValues);
-            }
-        }
-
-        private void RenderBoundHTMLAttributes(
-            IList<KeyValuePair<string, Chunk>> chunkAttributes,
-            string tagHelperTypeName,
-            string tagHelperVariableName,
-            IEnumerable<TagHelperAttributeDescriptor> attributeDescriptors,
-            Dictionary<string, string> htmlAttributeValues)
-        {
-            // Track dictionary properties we have confirmed are non-null.
-            var confirmedDictionaries = new HashSet<string>(StringComparer.Ordinal);
-
-            // First attribute wins, even if there are duplicates.
-            var distinctAttributes = chunkAttributes.Distinct(KeyValuePairKeyComparer.Default);
-
-            // Go through the HTML attributes in source order, assigning to properties or indexers as we go.
-            foreach (var attributeKeyValuePair in distinctAttributes)
-            {
-                var attributeName = attributeKeyValuePair.Key;
-                var attributeValueChunk = attributeKeyValuePair.Value;
-                if (attributeValueChunk == null)
+                // Execution contexts and thrown errors are a runtime feature.
```

Comment:
`... and thrown errors ...` confuses rather than helps. Suggest `... and throwing errors for null dictionary properties are runtime features.`